File: C/Livros/Fundamentos da Programação de Computadores/6. Vetor/Exercícios Resolvidos/Ex2.c
/*
Uma pequena loja de artesanato possui apenas um vendedor e comercializa dez tipos de objetos. O
vendedor recebe, mensalmente, salário de R$ 545,00, acrescido de 5% do valor total de suas vendas.
O valor unitário dos objetos deve ser informado e armazenado em um vetor; a quantidade vendida de
cada peça deve ficar em outro vetor, mas na mesma posição. Crie um programa que receba os preços
e as quantidades vendidas, armazenando-os em seus respectivos vetores (ambos com tamanho dez).
Depois, determine e mostre:
■■ um relatório contendo: quantidade vendida, valor unitário e valor total de cada objeto. Ao final,
deverão ser mostrados o valor geral das vendas e o valor da comissão que será paga ao vendedor; e
■■ o valor do objeto mais vendido e sua posição no vetor (não se preocupe com empates)
*/
#include <stdio.h>

int main()
{
    float preco[10];
    int qtsVendida[10];
    float valorGeral = 0;
    float maiorValor = 0;
    int posicao;

    // le preco e quantidade vendida
    for (int i = 0; i < 10; i++)
    {
        scanf("%f%d", &preco[i], &qtsVendida[i]);

        valorGeral += (preco[i] * qtsVendida[i]);

        if (qtsVendida[i] > maiorValor)
        {
            maiorValor = preco[i];
            posicao = i;
        } // end if

    } // end for

    // imprime relatorio
    for (int i = 0; i < 10; i++)
    {

        printf("Quantidade vendida: %d Valor: %.2f Valor Total %.2f\n", qtsVendida[i], preco[i], (qtsVendida[i] * preco[i]));

    } // end for

    printf("Valor geral vendas: %.2f Comissao: %.2f\n", valorGeral, (0.05f * valorGeral));
    printf("Valor objeto mais vendido: %.2f Posicao: %d\n", maiorValor, posicao);

    return 0;
} // end main

File: Java/Livros/Fundamentos da programação de computadores/Capítulo 6 Vetor/Ex2.java
import java.util.Scanner;

public class Ex2 {
    static public void main(String[] args) {

        // declara vetor
        int[] quantidade = new int[10];
        float[] preco = new float[10];

        // cria objeto scanner
        Scanner dado = new Scanner(System.in);

        // leitura preco e quantidade
        for (int i = 0; i < preco.length; i++) {

            quantidade[i] = dado.nextInt();
            preco[i] = dado.nextFloat();

        }

        float salario = 545;

        // Relatório e preco da comissao
        for (int i = 0; i < preco.length; i++) {

            System.out.println("Preco Unitario " + preco[i] + " Quantidade: " + quantidade[i] + " Valor Total: "
                    + (preco[i] * quantidade[i]));
            System.out.println("Comisao: " + ((preco[i] * quantidade[i]) * 0.05f));

        }

        // falta objeto mais vendido e posicao no vetor
        int maisVendido = 0;
        int maiorQuantidade = quantidade[0];

        // trata mais vendido e posicao no vetor
        for (int i = 0; i < preco.length; i++) {
            if (quantidade[i] > maiorQuantidade) {
                maiorQuantidade = quantidade[i];
                maisVendido = i;
            }

        }

        System.out.println("Mais vendido: " + maisVendido + " Quantidade: " + maiorQuantidade);

    }// end main
}
